src/test/test_marshal.cpp

```
@@ -605,6 +605,26 @@ TEST_CASE("disasm_marshal_Mem", "[disasm][marshal]") {
     }
 }
 
+TEST_CASE("unmarshal ST sign-extends the immediate", "[disasm][marshal]") {
+    // A BPF_ST store assigns its signed 32-bit immediate to the target location.
+    // For a 64-bit (DW) store the kernel sign-extends, so -1 becomes
+    // 0xFFFFFFFFFFFFFFFF, not 0x00000000FFFFFFFF. Model it the same way, matching
+    // the ALU-immediate path (getBinValue).
+    const ProgramInfo info{.platform = &g_ebpf_platform_linux,
```

Cppcheck (2.21.0)

[style] The function 'repair_potential' is never used.

(unusedFunction)